File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
import math

def divisor(num):
    if (num == 1):
        print(num, 1)
        return 1
    sq = math.sqrt(num)
    result = 0
    for i in range(1, int(sq) + 1):
        if (num % i) == 0:
            result += 1
    if (num % sq) == 0:
        result = result * 2 - 1
    else :
        result *= 2
    return result

def get_weapon(num, limit, power):
    weapon = divisor(num)
    if (weapon > limit):
        weapon = power
    return weapon
    
def solution(number, limit, power):
    answer = 0
    for i in range(1, number + 1):
        answer += get_weapon(i, limit, power)
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int getWeapon(int num, int limit, int power) {
    int result = 0;
    for (int i = 1; i * i <= num; i++) {
        if ((i * i) == num) result += 1;
        else if (num % i == 0) result += 2;
    }
    if (result > limit) return power;
    return result;
}

int solution(int number, int limit, int power) {
    int answer = 0;
    for (int i = 1; i <= number; i++) {
        int cnt = getWeapon(i, limit, power);
        answer += cnt;
    }
    return answer;
}